server/src/main/java/org/opensearch/search/pipeline/SearchPipelineService.java

```diff
@@ -360,7 +362,7 @@ static ClusterState innerDelete(DeleteSearchPipelineRequest request, ClusterStat
         return newState.build();
     }
 
-    public PipelinedRequest resolvePipeline(SearchRequest searchRequest) {
+    public PipelinedRequest resolvePipeline(SearchRequest searchRequest, IndexNameExpressionResolver indexNameExpressionResolver) {
```

Comment:
We are changing a public method here without providing a backward compatible overload. It's not clear whether this file is API or internal.  I did do a brief search of all our repos and don't see it called anywhere else, but we should probably either make this compatible with an overload or add the opensearch-internal annotation to this class.

Reply:
@froh wdyt of adding `@opensearch.internal` to this class?

Reply:
Yes -- we should label it `@opensearch.internal`. I didn't know enough about the annotations when I added it.